### PR TITLE
Fix/issue-129 test project

### LINKED ISSUE #129: Proposal: a "real project" for testing the command line interface
### Feature description

It would be nice if it was easier to test the output of running `status` within the `signac-flow` codebase. I currently have to test new changes on "live" projects, which is generally bad practice. Including this in the codebase would make it easy for anyone doing dev work to test out changes to the code without having to keep a separate test project up-to-date.

### Proposed solution

I propose we add a directory within the `tests/` directory that contains a script that sets up a "real-life" project that has a somewhat complex workflow. A script would also be included to call all `status` with all (or at least a lot) permutations of its various arguments to see how any changes to the output appear on screen.

### PR DESCRIPTION
**Pull request by tcmoore3**
Fixes #129 

## Description
Adds a test project to test how status looks for a project with a somewhat complicated workflow.

## Motivation and Context
See issue #129 

## Types of Changes
- [x] New feature

<sup>1</sup>The change breaks (or has the potential to break) existing functionality.

## Checklist:

- [x] I am familiar with the [**Contributing Guidelines**](https://github.com/glotzerlab/signac-flow/blob/master/CONTRIBUTING.md).
- [x] I agree with the terms of the [**Contributor Agreement**](https://github.com/glotzerlab/signac-flow/blob/master/ContributorAgreement.md).
- [x] My name is on the [list of contributors](https://github.com/glotzerlab/signac-flow/blob/master/contributors.yaml).
- [x] My code follows the [code style guideline](https://github.com/glotzerlab/signac-flow/blob/master/CONTRIBUTING.md#code-style) of this project.
- [x] The changes introduced by this pull request are covered by existing or newly introduced tests.

If necessary:
- [x] I have updated the API documentation as part of the package doc-strings.
- [x] I have created a separate pull request to update the [framework documentation](https://docs.signac.io/) on [signac-docs](https://github.com/glotzerlab/signac-docs) and linked it here.
- [x] I have updated the [changelog](https://github.com/glotzerlab/signac-flow/blob/master/changelog.txt).

### CODE REVIEW

**Comment by vyasr** [comment]
Regarding @csadorf's second point, we could just provide two separate functions for generating data, one that's more complex and one that's simpler.

File: flow/testing.py [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+# This software is licensed under the BSD 3-Clause License.
+"""Initialize a project for testing purposes
+
+"""
+import signac
+from signac.testing import init_jobs
+from .template import init
+
+
+def make_project(alias='project', root=None, **kwargs):
```
Comment (tcmoore3) [inline flow/testing.py]: Should this function return the project?

File: changelog.txt [hunk 2]
```diff
@@ -14,6 +14,7 @@ next
 - Allow the provision of multiple operation-functions to the ``pre.after`` and ``*.copy_from`` conditions (#120).
 - Add option to specify the operation execution order (#121).
 - Fix erroneous line in the torque submission template (#126).
+- Add a test project to test status output in tests/
```
Comment (tcmoore3) [inline changelog.txt]: Not current anymore, fix.

File: flow/testing.py (hunk 1 above)
Comment (tcmoore3) [reply flow/testing.py]: Yes I think it should.

Comment (tcmoore3) [reply flow/testing.py]: Fixed

File: changelog.txt (hunk 2 above)
Comment (tcmoore3) [reply changelog.txt]: Fixed

**Comment by csadorf** [comment]
Merge after https://github.com/glotzerlab/signac/pull/215

**Pull request merged**